sensor/common/registry/registry_store.go

```diff
@@ -201,3 +213,68 @@ func (rs *Store) GetGlobalRegistryForImage(image *storage.ImageName) (registryTy
 
 	return nil, errors.Errorf("unknown image registry: %q", reg)
 }
+
+// SetDelegatedRegistryConfig sets a new delegated registry config for use in determining
+// if a particular image is from a registry that should be accessed local to this cluster
+func (rs *Store) SetDelegatedRegistryConfig(config *central.DelegatedRegistryConfig) {
+	rs.delegatedRegistryConfigMutex.Lock()
+	defer rs.delegatedRegistryConfigMutex.Unlock()
+	rs.delegatedRegistryConfig = config
+}
+
+// IsLocal determines if an image is from a registry that should be accessed
+// local to this secured cluster.  Always returns true for image registry's that have
+// been added via AddClusterLocalRegistryHost
+func (rs *Store) IsLocal(image *storage.ImageName) bool {
+	if image == nil {
+		return false
+	}
+
+	if rs.hasClusterLocalRegistryHost(image.GetRegistry()) {
+		// This host is always cluster local irregardless of the DelegatedRegistryConfig (ie: OCP internal registry).
+		return true
+	}
+
+	rs.delegatedRegistryConfigMutex.RLock()
+	defer rs.delegatedRegistryConfigMutex.RUnlock()
+
+	config := rs.delegatedRegistryConfig
+	if config == nil || config.EnabledFor == central.DelegatedRegistryConfig_NONE {
+		return false
+	}
+
+	if config.EnabledFor == central.DelegatedRegistryConfig_ALL {
+		return true
+	}
+
+	// if image matches a delegated registry prefix, it is local
+	imageFullName := urlfmt.TrimHTTPPrefixes(image.GetFullName())
+	for _, r := range config.Registries {
+		regPath := urlfmt.TrimHTTPPrefixes(r.RegistryPath)
+		if strings.HasPrefix(imageFullName, regPath) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// AddClusterLocalRegistryHost adds host to an internal set of hosts representing
+// registries that are only accessible from this cluster. These hosts will factored
+// into IsLocal decisioning. Is OK to call repeatedly for the same host
+func (rs *Store) AddClusterLocalRegistryHost(host string) {
+	rs.clusterLocalRegistryHostsMutex.Lock()
+	defer rs.clusterLocalRegistryHostsMutex.Unlock()
+
+	trimmed := urlfmt.TrimHTTPPrefixes(host)
+	rs.clusterLocalRegistryHosts.Add(trimmed)
+
+	log.Debugf("Added cluster local registry host %q", trimmed)
+}
+
+func (rs *Store) hasClusterLocalRegistryHost(host string) bool {
+	rs.clusterLocalRegistryHostsMutex.RLock()
+	defer rs.clusterLocalRegistryHostsMutex.RUnlock()
+
+	return rs.clusterLocalRegistryHosts.Contains(urlfmt.TrimHTTPPrefixes(host))
```

Comment:
is it possible for one image to use https://my-reg and another to use http://my-reg, and they are meant to be different? Guess that wouldn't make a lot of sense, but just want to bring it up

Reply:
I suppose its technically possible for requests to 443 and 80 to be routed to different back-ends, but for the cluster local registries (ie: OCP internal registry) thats not an option. Removing the prefix is used elsewhere when matching a registry to an integration, essentially doing the same here.

Reply:
mentioned above, but let's move the prefix trimming outside the locked section